hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapReplaceAllMessageTask.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.map;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.MapReplaceAllCodec;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.internal.nio.Connection;
+import com.hazelcast.map.impl.MapEntryReplacingEntryProcessor;
+import com.hazelcast.map.impl.MapService;
+import com.hazelcast.map.impl.operation.MapOperationProvider;
+import com.hazelcast.query.Predicates;
+import com.hazelcast.security.permission.ActionConstants;
+import com.hazelcast.security.permission.MapPermission;
+import com.hazelcast.spi.impl.operationservice.OperationFactory;
+
+import java.security.Permission;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+public class MapReplaceAllMessageTask
+        extends AbstractMapAllPartitionsMessageTask<MapReplaceAllCodec.RequestParameters> {
+
+    private BiFunction function;
+
+    public MapReplaceAllMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected OperationFactory createOperationFactory() {
+        MapOperationProvider operationProvider = getOperationProvider(parameters.name);
+        return operationProvider.createPartitionWideEntryWithPredicateOperationFactory(parameters.name,
+           new MapEntryReplacingEntryProcessor(function), Predicates.alwaysTrue());
+    }
+
+    @Override
+    protected Object reduce(Map<Integer, Object> map) {
+        return null;
+    }
+
+    @Override
+    public String getServiceName() {
+        return MapService.SERVICE_NAME;
+    }
+
+    @Override
+    protected MapReplaceAllCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        MapReplaceAllCodec.RequestParameters parameters =  MapReplaceAllCodec.decodeRequest(clientMessage);
+        function = serializationService.toObject(parameters.function);
+        return parameters;
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return MapReplaceAllCodec.encodeResponse();
+    }
+
+    @Override
+    public Permission getRequiredPermission() {
+        return new MapPermission(getDistributedObjectName(), ActionConstants.ACTION_PUT);
```

Comment:
should it be `ACTION_MODIFY`?

Reply:
I see MapReplaceMessageTask has been using ACTION_PUT. Can you help me understand why do you think it can be ACTION_MODIFY ?